tests/Client/RollbackAttackTest.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace Tuf\Tests\Client;
+
+use Tuf\Exception\Attack\RollbackAttackException;
+use Tuf\Tests\ClientTestBase;
+use Tuf\Tests\TestHelpers\DurableStorage\TestStorage;
+
+/**
+ * Tests that server-side rollback attacks are detected.
+ */
+class RollbackAttackTest extends ClientTestBase
+{
+    /**
+     * {@inheritdoc}
+     */
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        // Always throw an exception if writing to client storage.
+        $this->clientStorage = new class () extends TestStorage {
+
+            /**
+             * {@inheritDoc}
+             */
+            public function write(string $name, string $data): void
+            {
+                throw new \LogicException("Unexpected attempt to change client storage.");
+            }
+
+            /**
+             * {@inheritDoc}
+             */
+            public function delete(string $name): void
```

Comment:
This is doing the same thing that TestStorage is doing in HEAD. It overrides `delete()` to throw an exception. I merely decided not to repeat the exception itself, and just call `write()` internally.

Reply:
Since it still just 1 line it seems like would just be clearer to throw the exception in both places. I pretty sure I won't have asked the question if the exception was in both places

It doesn't seem worth adding comment instead of throwing the exception in 2 places

Reply:
Fixed!